snappi_ixnetwork/snappi_api.py

```diff
@@ -284,26 +308,91 @@ def _protocols_exists(self):
-                              .Ethernet.find())
+                eth_dev = len(
+                    self._ixnetwork.Topology.find()
+                    .DeviceGroup.find()
+                    .Ethernet.find()
+                )
                 ethv4v6_dev_count = ethv4v6_dev_count + eth_dev
                 if eth_dev > 0:
-                    v4_dev = len(self._ixnetwork.Topology.find()
-                                 .DeviceGroup.find()
-                                 .Ethernet.find().Ipv4.find())
+                    v4_dev = len(
+                        self._ixnetwork.Topology.find()
+                        .DeviceGroup.find()
+                        .Ethernet.find()
+                        .Ipv4.find()
+                    )
                     ethv4v6_dev_count = ethv4v6_dev_count + v4_dev
-                    v6_dev = len(self._ixnetwork.Topology.find()
-                                 .DeviceGroup.find()
-                                 .Ethernet.find().Ipv6.find())
+                    v6_dev = len(
+                        self._ixnetwork.Topology.find()
+                        .DeviceGroup.find()
+                        .Ethernet.find()
+                        .Ipv6.find()
+                    )
                     ethv4v6_dev_count = ethv4v6_dev_count + v6_dev
-        if (total_dev > ethv4v6_dev_count):
+        if total_dev > ethv4v6_dev_count:
             return True
         else:
             return False
 
+    def set_control_state(self, payload):
+        try:
+            control_option = payload.choice
+            control_obj = getattr(payload, control_option)
+            control_choice = control_obj.get("choice")
+            request_payload = getattr(control_obj, control_choice)
+            self._connect()
+            if control_option == "port":
+                if control_choice == "capture":
+                    self.capture.set_capture_state(request_payload)
+                elif control_choice == "link":
+                    self.vport.set_link_state(request_payload)
+            elif control_option == "protocol":
+                if control_choice == "all":
+                    self.ngpf.set_protocol_state(request_payload)
+                elif control_choice == "route":
+                    self.ngpf.set_route_state(request_payload)
+                elif control_choice == "lacp":
+                    self.ngpf.set_device_state(request_payload)
+            elif control_option == "traffic":
+                self.traffic_item.transmit(request_payload)
+        except Exception as err:
+            raise SnappiIxnException(err)
+        return self._request_detail()
+
+    def set_control_action(self, payload):
+        try:
+            control_option = payload.choice
+            control_obj = getattr(payload, control_option)
+            control_choice = control_obj.get("choice")
+            choice_obj = getattr(control_obj, control_choice)
+            if control_choice == "ipv4":
```

Comment:
add else check/ validation for validation if `control_choice` and `choice` will not supported